packages/graph/src/query.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  BaseQuery,
+  CeramicApi,
+  PageInfo as QueryPageInfo,
+  Pagination,
+  RunningStateLike,
+  StreamState,
+} from '@ceramicnetwork/common'
+import { ModelInstanceDocument } from '@ceramicnetwork/stream-model-instance'
+import { StreamID } from '@ceramicnetwork/streamid'
+import type {
+  Connection,
+  ConnectionArguments,
+  Edge,
+  PageInfo as RelayPageInfo,
+} from 'graphql-relay'
+
+export type ConnectionQuery = BaseQuery & ConnectionArguments
+export type IndexQuery = BaseQuery & Pagination
+
+export class RunningState implements RunningStateLike {
+  #id: StreamID
+  #state: StreamState
+
+  constructor(state: StreamState) {
+    this.#id = new StreamID(state.type, state.log[0].cid)
+    this.#state = state
+  }
+
+  get id(): StreamID {
+    return this.#id
+  }
+
+  get state(): StreamState {
+    return this.#state
+  }
+
+  get value(): StreamState {
+    return this.#state
+  }
+
+  next(): never {
+    throw new Error('Unsupported method')
+  }
+
+  pipe(): never {
+    throw new Error('Unsupported method')
+  }
+
+  subscribe(): never {
+    throw new Error('Unsupported method')
+  }
+}
+
+function createModelInstance(state: StreamState): ModelInstanceDocument {
+  if (state.type !== ModelInstanceDocument.STREAM_TYPE_ID) {
+    throw new Error(`Unexpected stream type: ${state.type}`)
+  }
+  return new ModelInstanceDocument(new RunningState(state), {})
+}
```

Comment:
@ukstv could you have a look at this when you get the chance and let me know if it matches what you had in mind for the client-side creation of MIDs from the `StreamState`?
If you have any suggestion on how to make it simpler or more performant please let me know!

Reply:
@PaulLeCam Basically, yes.

On http client side, we have an implementation of the `RunningState` called `Document`. Maybe instead of implementing `RunningState` here you could leverage the `Document` class.

It would look like `return new ModelInstanceDocument(new Document(state, apiURL, SYNC_INTERVAL)`.

Now as I write this, I realise we do not export `Document` instance from http-client package, along with the SYNC_INTERVAL configuration parameter.

So, if you have access to the http client, and you find the idea of leveraging existing `Document` class viable, please, let me know, and I will change the package to export the things suitable for you.

Reply:
if ^^ not, then LGTM.

Reply:
@ukstv thanks!
Yes I looked into using `Document` but as it's client-specific and currently our setup is based on the `CeramicApi` interface, it seemed to me it would be harder to integrate and test?
Is there a version of `Document` that wouldn't be specific to the HTTP client and could work with a `CeramicApi` interface directly please?